AttendanceSystem.Api/Controllers/SessionsController.cs

```diff
@@ -49,16 +49,24 @@ public async Task<IActionResult> GetSessionInfo([HttpTrigger(AuthorizationLevel.
         return new OkObjectResult(session);
     }
     
-    [Function( $"{nameof(SessionsController)}-{nameof(EditAttendance)}")]
-    public async Task<IActionResult> EditAttendance([HttpTrigger(AuthorizationLevel.Anonymous, "put", Route="courses/{courseId}/sessions/{sessionId:guid}/attendance")] HttpRequest req, string courseId, Guid sessionId, [FromBody] UpdateAttendanceContract contract)
+    [Function( $"{nameof(SessionsController)}-{nameof(EditStudentAttendance)}")]
+    public async Task<IActionResult> EditStudentAttendance([HttpTrigger(AuthorizationLevel.Anonymous, "put", Route="courses/{courseId}/sessions/{sessionId:guid}/attendance")] HttpRequest req, string courseId, Guid sessionId, [FromBody] UpdateAttendanceContract contract)
     {
         _logger.LogInformation("C# HTTP trigger function processed a request.");
 
-        await _attendanceService.SetAttendance(sessionId, contract.UserId, contract.Kind);
+        await _attendanceService.SetStudentAttendance(sessionId, contract.UserId, contract.Kind);
         
         return new NoContentResult();
     }
-    
+
+    [Function($"{nameof(SessionsController)}-{nameof(EditTeacherAttendance)}")]
+    public async Task<IActionResult> EditTeacherAttendance([HttpTrigger(AuthorizationLevel.Anonymous, "put", Route = "courses/{courseId}/sessions/{sessionId:guid}/teacherattendance")] HttpRequest req, string courseId, Guid sessionId, [FromBody] UpdateAttendanceContract contract)
+    {
+        _logger.LogInformation("C# HTTP trigger function processed a request.");
+        await _attendanceService.SetTeacherAttendance(sessionId, contract.UserId, contract.Kind);
```

Comment:
Like I said in my other comment, I think `SetTeacherAttendance` is not a good name, we could do something like `SetTeacherApproval` or `SetActualApproval`. (If we take the role-based way described above, we still have to have separate domain functions, so this is still relevant).